addons/quickmount/functions/fnc_addFreeSeatsActions.sqf

```diff
@@ -0,0 +1,227 @@
+    if (_preserveEngineOn) then {_target engineOn true};
+
+// moveIn right after moveOut doesn't work in MP for non-local vehicles, player just stays out
+// we have to wait some time (e.g. until player is out)
+// usually it takes 1 frame in SP and 3 frames in MP, so in MP looks a little lagging
+// also if unit isn't moved to new seat in 0.5 seconds, we move him back to his seat
+#define MOVE_IN(command) \
+    [ARR_3( \
+        {isNull objectParent (_this select 0)}, \
+        { \
+            params [ARR_2("_player","_moveInParams")]; \
+            _player command _moveInParams; \
+            [ARR_5( \
+                {!isNull objectParent _this}, \
+                { \
+                    if (GETVAR(_this,GVAR(damageBlocked),false)) then { \
+                        _this allowDamage true; \
+                        SETVAR(_this,GVAR(damageBlocked),nil); \
+                    }; \
+                }, \
+                _player, \
+                0.5, \
+                { \
+                    (_this getVariable QGVAR(moveBackParams)) call (_this getVariable QGVAR(moveBackCode)); \
+                    if (GETVAR(_this,GVAR(damageBlocked),false)) then { \
+                        _this allowDamage true; \
+                        SETVAR(_this,GVAR(damageBlocked),nil); \
+                    }; \
+                } \
+            )] call CBA_fnc_waitUntilAndExecute; \
+        }, \
+        [ARR_2(_player,_this select 2)] \
+    )] call CBA_fnc_waitUntilAndExecute;
+
+#define ICON_DRIVER    "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\role_driver_ca.paa"
+#define ICON_PILOT     "A3\ui_f\data\IGUI\Cfg\Actions\getinpilot_ca.paa"
+#define ICON_CARGO     "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\role_cargo_ca.paa"
+#define ICON_GUNNER    "A3\ui_f\data\IGUI\Cfg\Actions\getingunner_ca.paa"
+#define ICON_COMMANDER "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\role_commander_ca.paa"
+#define ICON_TURRET    "A3\ui_f\data\IGUI\RscIngameUI\RscUnitInfo\role_gunner_ca.paa"
+#define ICON_FFV       "A3\ui_f\data\IGUI\Cfg\CrewAimIndicator\gunnerAuto_ca.paa"
+
+scopeName "main";
+
+params ["_vehicle", "_player"];
+
+private _vehicleConfig = configFile >> "CfgVehicles" >> typeOf _vehicle;
+private _isInVehicle = _player in _vehicle;
+private _fullCrew = fullCrew [_vehicle, "", true];
+
+private ["_driverCompartments", "_isDriverIsolated", "_cargoCompartments", "_cargoCompartmentsLast", "_compartment"];
+
+if (_isInVehicle) then {
+    _driverCompartments = (_vehicleConfig >> "driverCompartments") call BIS_fnc_getCfgData;
+    // Air class by default has driverCompartments=0 and cargoCompartments[]={0}, so we have to disable them
+    _isDriverIsolated = _driverCompartments isEqualTo 0 && {_vehicle isKindOf "Air"};
+    TO_STRING(_driverCompartments);
+
+    _cargoCompartments = getArray (_vehicleConfig >> "cargoCompartments");
+    {
+        if !(_x isEqualType "") then {
+            _cargoCompartments set [_forEachIndex, format ["Compartment%1", _x]];
```

Comment:
Look how it was with `apply`: https://github.com/acemod/ACE3/pull/5745/commits/9e17c18ac404f18f61bd36a94bcc06c5113c3604#diff-ded96a376761ae515c888f369fb06b23R55
I changed it to `forEach` only last days because IMHO it looks better and clearer.

Macro sets variable not array member so I can't use it here.

Reply:
```
_cargoCompartments = _cargoCompartments apply {[format ["Compartment%1", _x], -x] select (_x isEqualType "")}
```

Reply:
will test in a week

Reply:
`apply` is slower (even without intermediate `cargoCompartments` assignment) and less clear :hand:

Reply:
Don't understand how it can be. If you are doing the same thing, but with less commands and less variables (forEachIndex) it shouldn't possibly be slower.

Reply:
0.0083 ms
```
private _cargoCompartments = getArray (configFile >> "CfgVehicles" >> typeOf cursorobject >> "cargoCompartments"); 
{ 
    if !(_x isEqualType "") then { 
        _cargoCompartments set [_forEachIndex, format ["Compartment%1", _x]]; 
    }; 
} forEach _cargoCompartments; 
_cargoCompartments 
```
vs
0.0108 ms
```
private _cargoCompartments = getArray (configFile >> "CfgVehicles" >> typeOf cursorobject >> "cargoCompartments") apply {[format ["Compartment%1", _x], _x] select (_x isEqualType "")}; 
_cargoCompartments 
```
when `cursorObject` is `I_Truck_02_covered_F`

Reply:
Well I expected you read the message on github. Instead of in a notification E-Mail or wherever.